packages/enhanced-img/src/preprocessor.js

```diff
@@ -119,23 +127,35 @@ export function image(opts) {
 				}
 			}
 
+			/**
+			 * @type {Array<ReturnType<typeof update_element>>}
+			 */
+			const pending_ast_updates = [];
 			// TODO: switch to zimmerframe with Svelte 5
-			// @ts-ignore
-			await asyncWalk(ast.html, {
-				/**
-				 * @param {import('svelte/types/compiler/interfaces').TemplateNode} node
-				 */
-				async enter(node) {
-					if (node.type === 'Element') {
-						// Compare node tag match
-						if (node.name === 'enhanced:img') {
+
+			walk(
+				/** @type {import('svelte/compiler').AST.Root} */ (ast),
+				{},
+				{
+					_(_, { next }) {
+						next();
+					},
+					/** @param {import('svelte/compiler').AST.RegularElement} node */
+					// @ts-ignore
```

Comment:
I cannot for the life of me get this to understand me

Reply:
the trick is for the first argument to `walk` to be a union of all the node types _inside_ `Root` (this would be the type argument to `walk` if you were being explicit, but it's also inferred)